torch/csrc/distributed/rpc/process_group_agent.h

```diff
@@ -88,6 +87,30 @@ class ProcessGroupAgent : public RpcAgent {
     std::mutex mutex_;
   };
 
+  // TODO: this class should inherit from a MetricsTracker, and can be extended
+  // to track num_sends, recvs, average size of messages, etc.
+  struct AverageMetricsTracker {
+    const std::string key_;
+    float currentAverage_;
```

Comment:
misc: unless you have zillions of these, maybe better to use double/int64 rather than float/int32?

float32 precision isn't so high, can decay quickly in practice, especially if you have values that are far apart (e.g. below (x + (y-x)/ 1 million) might start not modifying the value much)

also: maybe consider making the representation {sum, count} rather than {average, count}?

Reply:
We may be able to do this, though if we change average to sum, there could be overflow issues more easily, depending on how long-running a processes that writes to these metrics are.